static-checks/rule-identifiers/test.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/python3
+
+import re
+from collections import defaultdict
+
+from lib import util, results, oscap, versions
+
+
+profile_references = {
+    # srg, disa, stigid@PRODUCT or controls file
+    'stig': [
+        'https://public.cyber.mil/stigs/downloads/?_dl_facet_stigs=operating-systems%2Cgeneral-purpose-os',  # noqa:E501
+        'https://public.cyber.mil/stigs/cci/',
+        'https://public.cyber.mil/stigs/downloads/?_dl_facet_stigs=operating-systems%2Cunix-linux',
+    ],
+    # ospp
+    'ospp': ['https://www.niap-ccevs.org/Profile/PP.cfm'],
+    # cis@PRODUCT or controls file
+    'cis': ['https://www.cisecurity.org/benchmark/red_hat_linux/'],
+    # anssi (controls file)
+    'anssi': ['https://cyber.gouv.fr/sites/default/files/document/linux_configuration-en-v2.pdf'],
+    # ism
+    'ism_o': ['https://www.cyber.gov.au/acsc/view-all-content/ism'],
+    # hipaa
+    'hipaa': ['https://www.gpo.gov/fdsys/pkg/CFR-2007-title45-vol1/pdf/CFR-2007-title45-vol1-chapA-subchapC.pdf'],  # noqa: E501
+}
+# Version specific references
+if versions.rhel.major == 8:
+    # pcidss
+    profile_references['pci-dss'] = ['https://www.pcisecuritystandards.org/documents/PCI_DSS_v3-2-1.pdf']  # noqa: E501
+elif versions.rhel.major == 9:
+    # pcidss
+    profile_references['pci-dss'] = ['https://www.pcisecuritystandards.org/documents/PCI_DSS_v4_0.pdf']  # noqa: E501
+
+
+# Parse from datastream all rules and all their references
+def get_rule_references():
+    references = defaultdict(set)
+    for frames, elements in oscap.parse_xml(util.get_datastream()):
+        if len(frames) < 3 or frames[-3:] != ['Group', 'Rule', 'reference']:
+            continue
+
+        rule, reference = elements[-2:]
+        # TODO: use str.removeprefix on python 3.9+
+        rule = re.sub('^xccdf_org.ssgproject.content_rule_', '', rule.get('id'))
+        reference = reference.get('href')
+        references[rule].add(reference)
+
+    return references
+
+
+profiles = oscap.Datastream(util.get_datastream()).profiles
+rule_references = get_rule_references()
+
+for profile, references in profile_references.items():
+    for rule in profiles[profile].rules:
+        for reference in references:
+            if reference in rule_references[rule]:
+                results.report('pass', f'{rule} ({profile})')
+            else:
+                results.report('fail', f'{rule} ({profile})', f'missing {reference} ({profile})')
```

Comment:
Same as above, plus the fact that mentioning profile again is probably just noise in the note (it's already part of the result name).
```suggestion
                results.report('fail', f'{profile}/{rule}', f"missing {reference}")
```